domain/crossmodelrelation/service/remoteapplication.go

```diff
@@ -126,6 +123,65 @@ func (s *Service) AddRemoteApplicationOfferer(ctx context.Context, applicationNa
 	return nil
 }
 
+// AddRemoteApplicationConsumer adds a new synthetic application representing
+// a remote relation on the consuming model, to this, the offering model.
+func (s *Service) AddRemoteApplicationConsumer(ctx context.Context, args AddRemoteApplicationConsumerArgs) error {
+	ctx, span := trace.Start(ctx, trace.NameFromFunc())
+	defer span.End()
+
+	// The synthetic application name is prefixed with "remote-" to avoid
+	// name clashes with local applications.
+	synthApplicationName := fmt.Sprintf("remote-%s", args.RemoteApplicationName)
```

Comment:
So this isn't correct. This will actually be the `ConsumerApplicationUUID`. The consumer doesn't pass the application name through. Currently, 3.6 passes `ApplicationToken` then at this call site, we remote all `-` to become the name.

In theory we can use this knowledge to our advantage, as we can reverse engineer the ApplicationToken as it's a uuid without hyphens.

See: https://github.com/juju/juju/blob/9523616c7213ac9c013e11c4094f7652aad38fe8/apiserver/facades/controller/crossmodelrelations/crossmodelrelations.go#L273

Reply:
Oh nice, I thought this was always going to be the app name. Done, and I renamed the field in the (service) struct.